### PR TITLE
lib-react-components: update .npmignore

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Ignore mock light curves and screenshots in `@zooniverse/react-components`. That removes ~500k of unused files from the published package.
- Update the changelog.
- Bump the version to 1.5.1.

## Package
lib-react-components

## How to Review
`npm publish --dry-run` to check which files are included in the published package.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out

## Maintenance
- [ ] If not from dependabot, the PR creator has described the update (major, minor, or patch version, changelog)

### CODE REVIEW

File: packages/lib-react-components/.npmignore [hunk 1]
```diff
@@ -9,3 +9,7 @@ test
 .editorconfig
 .travis.yml
 webpack.dist.js
+
+**/mockLightCurves/*
+**/README.md
```
Comment (goplayoutside3) [inline packages/lib-react-components/.npmignore:14]: Just to double check - If the Readme files are all excluded, will the Readme instructions at the root level be unintentionally excluded from this package's npm page? https://www.npmjs.com/package/@zooniverse/react-components

Comment (eatyourgreens) [reply packages/lib-react-components/.npmignore:14]: I think the Readme at the root level is always included with a package. This should just remove the Readme files from individual components in the distributed build.

Comment (goplayoutside3) [reply packages/lib-react-components/.npmignore:14]: I ran `npm publish --dry-run` for lib-react-components on this branch and the root level Readme is included which is good!

However, the dry run also includes the other Readmes such as `dist/cjs/helpers/withLayer/README.md`. Is that expected?

Comment (eatyourgreens) [reply packages/lib-react-components/.npmignore:14]: Oh, that's a bug.

Comment (eatyourgreens) [reply packages/lib-react-components/.npmignore:14]: Oh! 😳 
>The following paths and files are never ignored, so adding them to .npmignore is pointless:
>- package.json
>- README (and its variants)
>- CHANGELOG (and its variants)
>- LICENSE / LICENCE
  
https://docs.npmjs.com/cli/v7/using-npm/developers#keeping-files-out-of-your-package

**Pull request merged**